centreon/www/include/configuration/configObject/timeperiod/DB-Func.php

```diff
@@ -568,3 +386,245 @@ function createTimePeriodsExceptions(array $params): void
     $statement->bindValue(':timeperiod_id', (int) $params['timeperiod_id'], \PDO::PARAM_INT);
     $statement->execute();
 }
+
+// ----------------- API CALLS --------------------
+
+/**
+ * Create a new timeperiod form formData.
+ *
+ * @param array<mixed> $ret
+ *
+ * @return int|null
+ */
+function insertTimePeriodInAPI(array $ret = []): int|null
+{
+    global $form, $basePath;
+
+    $formData = $ret === [] ? $form->getSubmitValues() : $ret;
+
+    try {
+        $timeperiodId = insertTimeperiodByApi($formData, $basePath);
+
+        return $timeperiodId;
+    } catch (Throwable $th) {
+        echo "<div class='msg' align='center'>" . _($th->getMessage()) . '</div>';
```

Comment:
here you need to write a log with new methods of centreonLog to log the error to improve error handling ;) 
```suggestion
    } catch (Throwable $th) {
        CentreonLog::create()->error(
            logTypeId: CentreonLog::TYPE_BUSINESS_LOG,
            message: "Error while inserting timeperiod by api : {$th->getMessage()}",
            customContext: ['form_data' => $formData, 'base_path' => $basePath],
            exception: $th
        );
        
        echo "<div class='msg' align='center'>" . _($th->getMessage()) . '</div>';
```